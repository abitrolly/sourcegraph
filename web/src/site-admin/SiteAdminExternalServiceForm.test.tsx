jest.mock('../settings/DynamicallyImportedMonacoSettingsEditor', () => ({
    DynamicallyImportedMonacoSettingsEditor: () => 'DynamicallyImportedMonacoSettingsEditor',
}))

import * as H from 'history'
import React from 'react'
import { noop } from 'rxjs'
import { ExternalServiceKind } from '../../../shared/src/graphql/schema'
import { SiteAdminExternalServiceForm } from './SiteAdminExternalServiceForm'
import { mount } from 'enzyme'

describe('<SiteAdminExternalServiceForm />', () => {
    const baseProps = {
        history: H.createMemoryHistory(),
        isLightTheme: true,
        onSubmit: noop,
        onChange: noop,
        jsonSchema: { $id: 'json-schema-id' },
        editorActions: [],
    }

    test('create GitHub', () => {
        expect(
            mount(
                <SiteAdminExternalServiceForm
                    {...baseProps}
                    input={{
                        kind: ExternalServiceKind.GITHUB,
                        displayName: 'GitHub',
                        config: '{}',
                    }}
                    mode="create"
                    loading={false}
                />
            )
        ).toMatchSnapshot()
    })
    test('edit GitHub', () => {
        expect(
            mount(
                <SiteAdminExternalServiceForm
                    {...baseProps}
                    input={{
                        kind: ExternalServiceKind.GITHUB,
                        displayName: 'GitHub',
                        config: '{}',
                    }}
                    mode="create"
                    loading={false}
                />
            )
        ).toMatchSnapshot()
    })
    test('edit GitHub, loading', () => {
        expect(
            mount(
                <SiteAdminExternalServiceForm
                    {...baseProps}
                    input={{
                        kind: ExternalServiceKind.GITHUB,
                        displayName: 'GitHub',
                        config: '{}',
                    }}
                    mode="create"
                    loading={true}
                />
            )
        ).toMatchSnapshot()
    })
})
