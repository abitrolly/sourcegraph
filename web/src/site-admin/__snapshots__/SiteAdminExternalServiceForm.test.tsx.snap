// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteAdminExternalServiceForm /> create GitHub 1`] = `
<SiteAdminExternalServiceForm
  editorActions={Array []}
  history="[History]"
  input={
    Object {
      "config": "{}",
      "displayName": "GitHub",
      "kind": "GITHUB",
    }
  }
  isLightTheme={true}
  jsonSchema={
    Object {
      "$id": "json-schema-id",
    }
  }
  loading={false}
  mode="create"
  onChange={[Function]}
  onSubmit={[Function]}
>
  <Form
    className="external-service-form"
    onSubmit={[Function]}
  >
    <form
      className="external-service-form "
      onInvalid={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="font-weight-bold"
          htmlFor="e2e-external-service-form-display-name"
        >
          Display name:
        </label>
        <input
          autoComplete="off"
          autoCorrect="off"
          autoFocus={true}
          className="form-control"
          disabled={false}
          id="e2e-external-service-form-display-name"
          onChange={[Function]}
          required={true}
          spellCheck={false}
          type="text"
          value="GitHub"
        />
      </div>
      <div
        className="form-group"
      >
        <DynamicallyImportedMonacoSettingsEditor
          actions={Array []}
          canEdit={false}
          className="e2e-external-service-editor"
          height={350}
          history="[History]"
          isLightTheme={true}
          jsonSchema={
            Object {
              "$id": "json-schema-id",
            }
          }
          loading={false}
          onChange={[Function]}
          value="{}"
        >
          DynamicallyImportedMonacoSettingsEditor
        </DynamicallyImportedMonacoSettingsEditor>
        <p
          className="form-text text-muted"
        >
          <small>
            Use Ctrl+Space for completion, and hover over JSON properties for documentation.
          </small>
        </p>
      </div>
      <button
        className="btn btn-primary mb-3 e2e-add-external-service-button"
        disabled={false}
        type="submit"
      >
        Add repositories
      </button>
    </form>
  </Form>
</SiteAdminExternalServiceForm>
`;

exports[`<SiteAdminExternalServiceForm /> edit GitHub 1`] = `
<SiteAdminExternalServiceForm
  editorActions={Array []}
  history="[History]"
  input={
    Object {
      "config": "{}",
      "displayName": "GitHub",
      "kind": "GITHUB",
    }
  }
  isLightTheme={true}
  jsonSchema={
    Object {
      "$id": "json-schema-id",
    }
  }
  loading={false}
  mode="create"
  onChange={[Function]}
  onSubmit={[Function]}
>
  <Form
    className="external-service-form"
    onSubmit={[Function]}
  >
    <form
      className="external-service-form "
      onInvalid={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="font-weight-bold"
          htmlFor="e2e-external-service-form-display-name"
        >
          Display name:
        </label>
        <input
          autoComplete="off"
          autoCorrect="off"
          autoFocus={true}
          className="form-control"
          disabled={false}
          id="e2e-external-service-form-display-name"
          onChange={[Function]}
          required={true}
          spellCheck={false}
          type="text"
          value="GitHub"
        />
      </div>
      <div
        className="form-group"
      >
        <DynamicallyImportedMonacoSettingsEditor
          actions={Array []}
          canEdit={false}
          className="e2e-external-service-editor"
          height={350}
          history="[History]"
          isLightTheme={true}
          jsonSchema={
            Object {
              "$id": "json-schema-id",
            }
          }
          loading={false}
          onChange={[Function]}
          value="{}"
        >
          DynamicallyImportedMonacoSettingsEditor
        </DynamicallyImportedMonacoSettingsEditor>
        <p
          className="form-text text-muted"
        >
          <small>
            Use Ctrl+Space for completion, and hover over JSON properties for documentation.
          </small>
        </p>
      </div>
      <button
        className="btn btn-primary mb-3 e2e-add-external-service-button"
        disabled={false}
        type="submit"
      >
        Add repositories
      </button>
    </form>
  </Form>
</SiteAdminExternalServiceForm>
`;

exports[`<SiteAdminExternalServiceForm /> edit GitHub, loading 1`] = `
<SiteAdminExternalServiceForm
  editorActions={Array []}
  history="[History]"
  input={
    Object {
      "config": "{}",
      "displayName": "GitHub",
      "kind": "GITHUB",
    }
  }
  isLightTheme={true}
  jsonSchema={
    Object {
      "$id": "json-schema-id",
    }
  }
  loading={true}
  mode="create"
  onChange={[Function]}
  onSubmit={[Function]}
>
  <Form
    className="external-service-form"
    onSubmit={[Function]}
  >
    <form
      className="external-service-form "
      onInvalid={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="font-weight-bold"
          htmlFor="e2e-external-service-form-display-name"
        >
          Display name:
        </label>
        <input
          autoComplete="off"
          autoCorrect="off"
          autoFocus={true}
          className="form-control"
          disabled={true}
          id="e2e-external-service-form-display-name"
          onChange={[Function]}
          required={true}
          spellCheck={false}
          type="text"
          value="GitHub"
        />
      </div>
      <div
        className="form-group"
      >
        <DynamicallyImportedMonacoSettingsEditor
          actions={Array []}
          canEdit={false}
          className="e2e-external-service-editor"
          height={350}
          history="[History]"
          isLightTheme={true}
          jsonSchema={
            Object {
              "$id": "json-schema-id",
            }
          }
          loading={true}
          onChange={[Function]}
          value="{}"
        >
          DynamicallyImportedMonacoSettingsEditor
        </DynamicallyImportedMonacoSettingsEditor>
        <p
          className="form-text text-muted"
        >
          <small>
            Use Ctrl+Space for completion, and hover over JSON properties for documentation.
          </small>
        </p>
      </div>
      <button
        className="btn btn-primary mb-3 e2e-add-external-service-button"
        disabled={true}
        type="submit"
      >
        <LoadingSpinner
          className="icon-inline"
        >
          <div
            className="loading-spinner icon-inline"
          />
        </LoadingSpinner>
        Add repositories
      </button>
    </form>
  </Form>
</SiteAdminExternalServiceForm>
`;
